### ISSUE TITLE
Navigation doesn't scroll reliably on iOS

### ISSUE THREAD

**Issue by reddolan** [vmware/clarity#875]
```
[X] bug
[ ] feature request
[ ] enhancement
```

### Expected behavior
when opening the collapsed navigation on a phone, the long list of navigation items should scroll up and down

### Actual behavior
- from the Documentation pages
- click vertical ellipses to show offcanvas navigation
- try to scroll the navigation to the lower menu options
- the main content scrolls, the nav does not
- not sure if this is just isolated to the website or is a component-level issue
- I can get the nav to scroll but only after collapsing and expanding one of the nav headers
- it seems like it is a focus issue

### Reproduction of behavior
![ios-navigation-issue](https://user-images.githubusercontent.com/1858226/29224436-3184a0cc-7e7f-11e7-9a2d-1b9b8253222b.gif)

### Environment details
- Clarity version: 0.9.3
- OS and version: iOS 10.3.1
- Browser: iOS Safari, iOS Chrome

**Comment by bojan88** [comment]
This bug is affecting all touch devices. I noticed it on Chrome for Android.
The problem is with CSS - `overflow-y: auto` is set on hover on non-touch devices, but on touch devices it's set on click event.
I filed a pull request for this.

I created a gif if it's not clear from the comments.
![clarity-menu-bug](https://user-images.githubusercontent.com/1783133/29211302-8d2cafdc-7e98-11e7-818d-cf278cf25355.gif)
